### PR TITLE
introducing new version_range model

### PR DESCRIPTION
**Pull request by memsharded**
It is missing the parsing of different version ranges models like ``~1.0`` and ``1.*``, but to see more or less the structure of the solution

### CODE REVIEW

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

File: conans/model/recipe_ref.py [hunk 1]
```diff
@@ -16,6 +17,27 @@ def __init__(self, value):
         self._pre = None
         self._build = None
 
+    def bump(self, index):
+        if self._items is None:  # the indicator parse is needed is empty items
```
Comment (lasote) [inline conans/model/recipe_ref.py:29]:
```suggestion
        if self._items is None:  # the indicator parse is needed if empty items
```

File: conans/model/recipe_ref.py [hunk 2]
```diff
@@ -16,6 +17,27 @@ def __init__(self, value):
         self._pre = None
         self._build = None
 
+    def bump(self, index):
```
Comment (lasote) [inline conans/model/recipe_ref.py:20]:
```suggestion
    def bump(self, index):
        """
           Increments by 1 the version field at the specified index, setting to 0 the fields on the right.
           2.5 => bump(1) => 2.6
           1.5.7 => bump(0) => 2.0.0
        """
```

**Review by lasote — APPROVED** [review]
I would add a simple unit test for the bump.
Can be merged.

**Comment by memsharded** [comment]
> I would add a simple unit test for the bump.

I have realized that bumping is not completely clear for prereleases and build, I will check it and add tests

**Pull request merged**